scipy/stats/stats.py

```diff
@@ -393,10 +397,11 @@ def hmean(a, axis=0, dtype=None):
                 size = a.shape[0]
             else:
                 size = a.shape[axis]
-        return size / np.sum(1.0 / a, axis=axis, dtype=dtype)
+        with np.errstate(divide='ignore'):
+            return size / np.sum(1.0 / a, axis=axis, dtype=dtype)
     else:
         raise ValueError("Harmonic mean only defined if all elements greater "
-                         "than zero")
+                         "than or equal to zero")
```

Comment:
This `ValueError` is never hit by tests. It may be worth probing that with a test to solidify the behavior, especially given the insightful analysis by Warren in the cognate issue on negative value handling.

Reply:
> This ValueError is never hit by tests. It may be worth probing that with a test to solidify the behavior, especially given the insightful analysis by Warren in the cognate issue on negative value handling.

Test case added. The negative value case also covers the bug in the standard lib mentioned in the issue thread.